counsel.el

```diff
@@ -1256,7 +1256,7 @@ files in a project.")
   "Default defun to calculate `counsel--git-grep-count'."
   (if (eq system-type 'windows-nt)
       0
-    (read (shell-command-to-string "du -s"))))
```

Comment:
What makes this more efficient?

Reply:
@abo-abo Calling `du` directly instead of via the shell cuts out the middle man (the shell).

I just noticed that `shell-command-to-string` calls `process-file` (which inspects file handlers) instead of `call-process`.

It is possible to keep this behaviour without calling the shell, thus still halving subprocess overhead:

```el
(read (with-output-to-string
        (process-file "du" nil standard-output nil "-s")))
```

Obviously this overhead reduction is completely negligible compared to the execution time of `du`. My comment was more about avoiding the shell where it's not needed than making any noticeable optimisation.

Reply:
I agree that not piping it all through the shell is a bit more efficient.

On the other hand, using the shell is a lot more flexible:

- You can copy paste the command into an actual shell to test it
- You can add/remove flags more easily, since it's just in the shell format

Reply:
@abo-abo Those aren't the arguments in favour of the shell's flexibility that I would have made (think piping, file descriptors, logical operators...), but agreed, the shell is of course more flexible.